### ISSUE TITLE
Undo Last Action in Planning

### ISSUE THREAD

**Issue by SonPlaying** [Anuken/Mindustry-Suggestions#2690]
**Describe the content or mechanics you are proposing.**
The undo button in planning a build, which undoes the last action made in planning.

For example, I've put a long conveyor belt, then another long conveyor belt but this one is mistaken and unaligned. So i want to delete this conveyor belt. the undo button does the job for me by undoing the last action made; which is deleting that mistaken and unaligned conveyor belt only, and not the previous belt.

For PC Players, add a hotkey for the undo button.

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**
This feature will surely ease any frustrations when misplacing an object, (we are not talking about schematics), especially for mobile players. When someone mistakenly places something (in planning) they can undo the last thing placed by pressing the undo button on the corner, and not only by clicking or tapping on it(one by one if multiple objects are placed) to delete it.
Basically an undo button in planning a build.

(What i mean by planning is that time where you have to place your objects, before pressing the done button)

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
2. - [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
3. - [X] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by Hyrotrioskjan** [comment]
I don't understand.. you can just delete blocks while in planning, is it that hard?
![image](https://user-images.githubusercontent.com/84776228/125262023-0365c700-e32c-11eb-95de-2174f71fbc3b.png)
You don't need an undo button, simply just delete the mistaken plan and done. 
Either way I guess an undo button _would_ be useful but why add that when something close to that is already in the game?

**Comment by oinkymoo** [comment]
if your belts are squiggly you might need a lot more rectangles to avoid other structures

**Comment by Humanoid-X** [comment]
why would you undo a squiggly line, why whould you even realise that you are doing wrong AFTER planning that line

**Comment by SonPlaying** [comment]
> I don't understand.. you can just delete blocks while in planning, is it that hard?
> ![image](https://user-images.githubusercontent.com/84776228/125262023-0365c700-e32c-11eb-95de-2174f71fbc3b.png)
> You don't need an undo button, simply just delete the mistaken plan and done.
> Either way I guess an undo button _would_ be useful but why add that when something close to that is already in the game?

No, what i mean by "planning" is before clicking the checkmark (done button), that time where you place your objects. What i want is to undo the last object planned in that mode.

**Comment by SonPlaying** [comment]
> why would you undo a squiggly line, why whould you even realise that you are doing wrong AFTER planning that line

What i mean by planning is not the time after you've pressed the done button when the builder builds the plan. What I'm talking about is the time you place your objects
